ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -255,7 +255,23 @@ static bool sqlite_handle_begin(pdo_dbh_t *dbh)
 {
 	pdo_sqlite_db_handle *H = (pdo_sqlite_db_handle *)dbh->driver_data;
 
-	if (sqlite3_exec(H->db, "BEGIN", NULL, NULL, NULL) != SQLITE_OK) {
+	char *begin_statement;
+	switch (H->transaction_mode) {
+		case 0:
+		case PDO_SQLITE_ATTR_TRANSACTION_MODE_DEFERRED:
+			begin_statement = "BEGIN DEFERRED TRANSACTION";
+			break;
+		case PDO_SQLITE_ATTR_TRANSACTION_MODE_IMMEDIATE:
+			begin_statement = "BEGIN IMMEDIATE TRANSACTION";
+			break;
+		case PDO_SQLITE_ATTR_TRANSACTION_MODE_EXCLUSIVE:
+			begin_statement = "BEGIN EXCLUSIVE TRANSACTION";
+			break;
+		default:
+			begin_statement = "BEGIN";
```

Comment:
I think this is unreachable?

Reply:
It was probably because of this https://github.com/php/php-src/actions/runs/17460972563/job/49585352842?pr=19317. I switched to initializing the declaration now, should work as well.